extension/js/common/api/account-server.ts

```diff
@@ -20,19 +20,21 @@ export type UploadedMessageData = {
  *   whether FES is deployed on the customer domain or not.
  */
 export class AccountServer extends Api {
-  private readonly potentialCustomerUrlFes: ExternalService;
-  private readonly sharedTenantFes: ExternalService;
+  private readonly externalService: ExternalService;
 
   public constructor(private acctEmail: string) {
     super();
-    this.potentialCustomerUrlFes = new ExternalService(this.acctEmail);
-    this.sharedTenantFes = new ExternalService(this.acctEmail);
-    this.sharedTenantFes.url = SHARED_TENANT_API_HOST;
+    this.externalService = new ExternalService(this.acctEmail);
   }
 
+  public initialize = async () => {
+    if (!(await isCustomerUrlFesUsed(this.acctEmail))) {
+      this.externalService.url = SHARED_TENANT_API_HOST;
+    }
```

Comment:
Can we move this check to `constructor`? So we won't need to run it separately on each `AccountServer` initialisation. 

Then this code
```ts
const acctServer = new AccountServer(acctEmail);
await acctServer.initialize();
await acctServer.fetchAndSaveClientConfiguration();
```

Can be just 
```ts
await new AccountServer(acctEmail).fetchAndSaveClientConfiguration();
```
as it was before

Reply:
But as you know `initialize` is async functino and can't be placed in constructor.

Reply:
What about using static `init` function, like this
```ts
class AccountServer  {
  public static init = async (acctEmail: string) => {
    const acctServer = new AccountServer(acctEmail);
    await acctServer.initialize();
    return acctServer
  };
```

And then used this way
```ts
await AccountServer.init(acctEmail).fetchAndSaveClientConfiguration();
```

In this case we won't forget to run `initialize` for each `AccountServer` init

Reply:
It sounds good when AccountServier init was called from async funtion.
But in several places, we init AccountServer in constructor (like ComposeView consturctor and accountserver was used from other view modules).
For those places, we still need to call initalize in render function or so

Reply:
Got it 👍 Looks good now